internal/content/content.go

```diff
@@ -0,0 +1,382 @@
+	switch contentType {
+	case ContentTypeTest:
+		return Config{
+			Type:        ContentTypeTest,
+			Directory:   utils.TestsDir,
+			DisplayName: "test",
+		}
+	case ContentTypeSnippet:
+		return Config{
+			Type:        ContentTypeSnippet,
+			Directory:   utils.SnippetsDir,
+			DisplayName: "snippet",
+		}
+	default:
+		panic("unsupported content type")
+	}
+}
+
+func Push(ctx context.Context, fsys afero.Fs, contentType ContentType) error {
+	if err := flags.LoadConfig(fsys); err != nil {
+		return err
+	}
+
+	config := GetConfig(contentType)
+	
+	exists, err := afero.DirExists(fsys, config.Directory)
+	if err != nil || !exists {
+		return errors.Errorf("%s does not exist", config.Directory)
+	}
+
+	entries, err := afero.ReadDir(fsys, config.Directory)
+	if err != nil {
+		return err
+	}
+
+	locals := map[string]string{}
+	for _, e := range entries {
+		if e.IsDir() || !strings.HasSuffix(e.Name(), ".sql") {
+			continue
+		}
+		name := strings.TrimSuffix(e.Name(), ".sql")
+		bytes, _ := afero.ReadFile(fsys, filepath.Join(config.Directory, e.Name()))
+		locals[name] = string(bytes)
+	}
+
+	if len(locals) == 0 {
+		fmt.Printf("No %s files found to push\n", config.DisplayName)
+		return nil
+	}
+
+	remote, err := listRemote(ctx, contentType)
+	if err != nil {
+		return err
+	}
+
+	for name, sql := range locals {
+		if id, ok := remote[name]; ok {
+			if err := upsert(ctx, contentType, id, name, sql); err != nil {
+				fmt.Fprintf(utils.GetDebugLogger(), "failed to update %s %s: %v\n", config.DisplayName, name, err)
+			} else {
+				fmt.Printf("Updated %s %s\n", config.DisplayName, utils.Aqua(name))
+			}
+		} else {
+			if err := create(ctx, contentType, name, sql); err != nil {
+				fmt.Fprintf(utils.GetDebugLogger(), "failed to create %s %s: %v\n", config.DisplayName, name, err)
+			} else {
+				fmt.Printf("Created %s %s\n", config.DisplayName, utils.Aqua(name))
+			}
+		}
+	}
+
+	var deletions []uuid.UUID
+	var deletionNames []string
+	for name, id := range remote {
+		if _, ok := locals[name]; !ok {
+			deletions = append(deletions, id)
+			deletionNames = append(deletionNames, name)
+		}
+	}
+
+	if len(deletions) > 0 {
+		console := utils.NewConsole()
+		msg := fmt.Sprintf("Your remote project has %d %s(s) that don't exist locally:\n  %s\nDo you want to delete them from the remote project?",
+			len(deletions), config.DisplayName, strings.Join(deletionNames, "\n  "))
+		shouldDelete, err := console.PromptYesNo(ctx, msg, false)
+		if err != nil {
+			return err
+		}
+		if shouldDelete {
+			for _, id := range deletions {
+				utils.GetSupabase().V1ProjectDeleteSnippet(ctx, flags.ProjectRef, id)
+			}
+			fmt.Printf("Deleted %d obsolete %ss\n", len(deletions), config.DisplayName)
+		} else {
+			fmt.Printf("Skipped deletion of %d remote %ss\n", len(deletions), config.DisplayName)
+		}
+	}
+	return nil
+}
+
+func DownloadAll(ctx context.Context, fsys afero.Fs, contentType ContentType) error {
```

Comment:
I'd suggest breaking this up

- `ListSnippets(ctx context.Context, contentType ContentType)`
- `DownloadSnippets(ctx context.Context, snippetIds []uuid.UUID, fsys afero.Fs)`

There's no need to have a separate DownloadOne since we can use a single element array.